skyportal/handlers/api/shift.py

```diff
@@ -515,3 +517,144 @@ def delete(self, shift_id, user_id):
             action='skyportal/REFRESH_SHIFTS', payload={'shift_id': int(shift_id)}
         )
         return self.success()
+
+
+class ShiftSummary(BaseHandler):
+    # this handler has a get method that returns a summary of all the activity of shift users on skyportal for a given period
+    # it is used to generate a report
+
+    @auth_or_token
+    def get(self, shift_id=None):
+        """
+        ---
+        description: Get a summary of all the activity of shift users on skyportal for a given period
+        tags:
+          - shifts
+          - users
+          - sources
+          - gcn
+        parameters:
+          - in: path
+            name: start_date
+            required: true
+            schema:
+              type: string
+              format: date
+          - in: path
+            name: end_date
+            required: true
+            schema:
+              type: string
+              format: date
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+        """
+
+        start_date = self.get_argument("start_date", None)
+        end_date = self.get_argument("end_date", None)
+        if (start_date is None or end_date is None) and shift_id is None:
+            return self.error("Please provide start_date and end_date, or shift_id")
+
+        if start_date is not None and end_date is not None:
+            try:
+                start_date = datetime.datetime.strptime(start_date, "%Y-%m-%dT%H:%M:%S")
+                end_date = datetime.datetime.strptime(end_date, "%Y-%m-%dT%H:%M:%S")
+            except ValueError:
+                return self.error("Please provide valid start_date and end_date")
+            if start_date > end_date:
+                return self.error("Please provide start_date < end_date")
+            if start_date.date() > datetime.datetime.today().date():
```

Comment:
Why do we need `.date()` and not just comparing the datetime objects directly?

Reply:
Pyarrow provides offset aware dates and datetime non offset aware dates, so you cant compare them without using `.date()`

Reply:
I replaced the right part of the statement by its equivalent using arrow